org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLRename.java

```diff
@@ -191,23 +194,35 @@ private boolean isRenameTagName(DOMDocument document, DOMElement element, Positi
 			Range startTagPrefixRange = new Range(startTagStartPosition, startTagPrefixPosition);
 			Range endTagPrefixRange = new Range(endTagStartPosition, endTagPrefixPosition);
 
-			if (doesTagCoverPosition(startTagPrefixRange, endTagPrefixRange, position)) {// Element prefix rename
-				String prefix = element.getPrefix();
-				return renameElementNamespace(xmlDocument, element, prefix.length(), newText);
-			} else { // suffix rename without wiping namespace
-				String suffixName = element.getLocalName();
-				int suffixLength = suffixName.length();
-				Position startTagEndPosition = startTagRange.getEnd();
-				Position suffixStartPositionStart = new Position(startTagEndPosition.getLine(),
-						startTagEndPosition.getCharacter() - suffixLength);
-
-				Position endTagEndPosition = endTagRange.getEnd();
-				Position suffixEndPositionStart = new Position(endTagEndPosition.getLine(),
-						endTagEndPosition.getCharacter() - suffixLength);
-
-				Range suffixRangeStart = new Range(suffixStartPositionStart, startTagEndPosition);
-				Range suffixRangeEnd = new Range(suffixEndPositionStart, endTagEndPosition);
+			String suffixName = element.getLocalName();
+			int suffixLength = suffixName.length();
+			Position startTagEndPosition = startTagRange.getEnd();
+			Position suffixStartPositionStart = new Position(startTagEndPosition.getLine(),
+					startTagEndPosition.getCharacter() - suffixLength);
+
+			Position endTagEndPosition = endTagRange.getEnd();
+			Position suffixEndPositionStart = new Position(endTagEndPosition.getLine(),
+					endTagEndPosition.getCharacter() - suffixLength);
+
+			Range suffixRangeStart = new Range(suffixStartPositionStart, startTagEndPosition);
+			Range suffixRangeEnd = new Range(suffixEndPositionStart, endTagEndPosition);
+
+			if (newText.contains(":")) {
+				String[] tagParts = newText.split(":", 2);
+				List<TextEdit> tagTextEdits = new ArrayList<TextEdit>();
+				// Tag prefix rename
+				if (!prefix.equals(tagParts[0])) {
+					tagTextEdits.addAll(renameElementNamespace(xmlDocument, element, prefix.length(), tagParts[0]));
+				}
+				// Tag suffix/local name rename
+				if (!prefix.equals(tagParts[1])) {
```

Comment:
Should this be `!suffixName.equals(tagParts[1])`.

Take the case of `<ns:nos>` and attempt to rename to `<ns:ns>`. Nothing happens.